test/Extism.Sdk/BasicTests.cs

```diff
@@ -168,4 +168,26 @@ public void HostFunctionsWithMemory()
         var response = plugin.Call("run_test", Encoding.UTF8.GetBytes("Frodo"));
         Encoding.UTF8.GetString(response).ShouldBe("HELLO FRODO!");
     }
+
+    [Fact]
+    public void LogLevel()
+    {
+        var tempFile = Path.GetTempFileName();
+        Plugin.SetLogFile(tempFile, Native.LogLevel.Warn);
+        using (var plugin = Helpers.LoadPlugin("log.wasm"))
+        {
+            plugin.Call("run_test", Array.Empty<byte>());
+        }
+
+        // HACK: tempFile gets locked by the Extism runtime 
```

Comment:
@zshipko is there any way to make the logger lock less restrictive? Currently I can't read the file even after I dispose the Plugin

Reply:
I'm not sure what kind of locking is happening here but it seems kind of surprising. I'll have to read through the log4rs code, which is handling that log file.

One thing worth noting is that the logging is a global configuration so disposing the Plugin wouldn't have any effect. Maybe there is some configuration about when to flush to disk that we could update? 

I also want to take a look at the logging situation this week and see if there are any paths for improvement.